library/core/src/ptr/non_null.rs

```diff
@@ -1451,6 +1455,9 @@ impl<T> NonNull<[T]> {
     #[inline]
     #[must_use]
     #[unstable(feature = "slice_ptr_get", issue = "74265")]
+    #[rustc_const_unstable(feature = "slice_ptr_get", issue = "74265")]
```

Comment:
The errors are due to this line. It was removed in https://github.com/model-checking/verify-rust-std/commit/8fe8f562171f4b696234c1313a25418878c1b885#diff-0d8566cc1e4354ac54927b27a9e6c5e27b8f289844c8346ef7822a6cb4a06569L1548.

Reply:
@zhassan-aws I just merged the updates and got these error. It seems like the error is now from the line `#[unstable(feature = "slice_ptr_get", issue = "74265")]`

```rust
Kani Rust Verifier 0.56.0 (standalone)
error: const function that might be (indirectly) exposed to stable cannot use `#[feature(slice_ptr_get)]`
    --> /Users/owo/Desktop/verify-rust-std/library/core/src/ptr/non_null.rs:1502:9
     |
1502 |         self.as_non_null_ptr().as_ptr()
     |         ^^^^^^^^^^^^^^^^^^^^^^
     |
     = help: mark the callee as `#[rustc_const_stable_indirect]` if it does not itself require any unsafe features
help: if the caller is not (yet) meant to be exposed to stable, add `#[rustc_const_unstable]` (this is what you probably want to do)
     |
1500 +     #[rustc_const_unstable(feature = "...", issue = "...")]
1501 |     #[ensures(|result: &*mut T| *result == self.pointer as *mut T)]
     |
help: otherwise, as a last resort `#[rustc_allow_const_fn_unstable]` can be used to bypass stability checks (this requires team approval)
     |
1500 +     #[rustc_allow_const_fn_unstable(slice_ptr_get)]
1501 |     #[ensures(|result: &*mut T| *result == self.pointer as *mut T)]
     |

error: const function that might be (indirectly) exposed to stable cannot use `#[feature(slice_ptr_get)]`
   --> /Users/owo/Desktop/verify-rust-std/library/core/src/ffi/c_str.rs:536:9
    |
536 | /         unsafe { NonNull::new_unchecked(&self.inner as *const [c_char] as *mut [c_char]) }
537 | |             .as_non_null_ptr()
    | |______________________________^
    |
    = help: mark the callee as `#[rustc_const_stable_indirect]` if it does not itself require any unsafe features
help: if the caller is not (yet) meant to be exposed to stable, add `#[rustc_const_unstable]` (this is what you probably want to do)
    |
533 +     #[rustc_const_unstable(feature = "...", issue = "...")]
534 |     const fn as_non_null_ptr(&self) -> NonNull<c_char> {
    |
help: otherwise, as a last resort `#[rustc_allow_const_fn_unstable]` can be used to bypass stability checks (this requires team approval)
    |
533 +     #[rustc_allow_const_fn_unstable(slice_ptr_get)]
534 |     const fn as_non_null_ptr(&self) -> NonNull<c_char> {
    |

error: Failed to execute cargo (exit status: 101). Found 2 compilation errors.
```

Reply:
Please remove this line.

Reply:
I removed it in https://github.com/model-checking/verify-rust-std/pull/126/commits/152d68ef1385544dd90c9314fd9d3639fbd2629c

Reply:
Thanks

Reply:
Thanks! So I ran all harnesses in `non_null.rs` locally with the command
```
./scripts/run-kani.sh --kani-args "--harness ptr::non_null::verify::non_null_check"
``` 
And saw in the results there are a few failed checks but in the end the verification successful which I'm confused with:
```
Checking harness ptr::non_null::verify::non_null_check_offset_from...

VERIFICATION RESULT:
 ** 1 of 271 failed (4 unreachable)
Failed Checks: attempt to subtract with overflow
 File: "/Users/owo/Desktop/verify-rust-std/library/core/src/ptr/non_null.rs", line 812, in ptr::non_null::NonNull::<i32>::offset_from::{closure#2}

VERIFICATION:- SUCCESSFUL (encountered one or more panics as expected)
Verification Time: 0.6992793s
......
Checking harness ptr::non_null::verify::non_null_check_is_aligned_to_no_power_two...

VERIFICATION RESULT:
 ** 1 of 18 failed
Failed Checks: This is a placeholder message; Kani doesn't support message formatted at runtime
 File: "/Users/owo/Desktop/verify-rust-std/library/core/src/ptr/const_ptr.rs", line 1526, in ptr::const_ptr::<impl *const i32>::is_aligned_to

VERIFICATION:- SUCCESSFUL (encountered one or more panics as expected)
Verification Time: 0.07415467s

Checking harness ptr::non_null::verify::non_null_check_align_offset_negative...

VERIFICATION RESULT:
 ** 1 of 310 failed (4 unreachable)
Failed Checks: This is a placeholder message; Kani doesn't support message formatted at runtime
 File: "/Users/owo/Desktop/verify-rust-std/library/core/src/ptr/non_null.rs", line 1311, in ptr::non_null::NonNull::<i8>::align_offset::{closure#2}::{closure#1}

VERIFICATION:- SUCCESSFUL (encountered one or more panics as expected)
Verification Time: 0.83871216s

Complete - 48 successfully verified harnesses, 0 failures, 48 total.
```

Reply:
@QinyuanWu when a function is annotated with `#[kani::should_panic]`, failed assertions do not cause verification to fail. See https://model-checking.github.io/kani/reference/attributes.html#kanishould_panic for an explanation.

Reply:
Ohhhhh that's why :0000 Sorry I missed that. Then I think @Dhvani-Kapadia can resolve the conversations and we are ready to merge!